### PR TITLE
V8 for win32

### PR DESCRIPTION
**Pull request by CedricGuillemet**

### CODE REVIEW

File: Core/Node-API/CMakeLists.txt [hunk 1]
```diff
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
 
             set(LINK_LIBRARIES ${LINK_LIBRARIES}
                 PUBLIC "${V8_ANDROID_DIR}/jni/${ANDROID_ABI}/libv8android.so")
+        elseif(WIN32)
+            set_cpu_platform_arch()
+            set(V8_VERSION "8.4.371.15")
+            #install_v8_nuget(${V8_VERSION} ${CPU_ARCH})
```
Comment (bghgary) [inline Core/Node-API/CMakeLists.txt:101]: Remove?

File: Core/Node-API/CMakeLists.txt [hunk 2]
```diff
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
...
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icudtl.dat"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icui18n.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icuuc.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8_libbase.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8_libplatform.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/zlib.dll")
+
+            # only 1 imported location per library -> Adding 1 library per file
+            foreach(V8FILE ${V8_DIST} )
```
Comment (bghgary) [inline Core/Node-API/CMakeLists.txt:126]: nit:
```suggestion
            foreach(V8FILE ${V8_DIST})
```

File: Core/Node-API/CMakeLists.txt [hunk 3]
```diff
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
...
+                get_filename_component(V8FILE_NAME "${V8FILE}" NAME_WE)
+                add_library("v8::${V8FILE_NAME}" SHARED IMPORTED)
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8_libbase.dll.lib")
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_LOCATION ${V8FILE})
+            endforeach()
+
+            set(LINK_LIBRARIES v8 v8::icudtl v8::icui18n v8::icuuc v8::v8 v8::v8_libbase v8::v8_libplatform v8::zlib)
+            
+            if (CPU_ARCH STREQUAL "x64")
```
Comment (bghgary) [inline Core/Node-API/CMakeLists.txt:135]: nit:
```suggestion
            if(CPU_ARCH STREQUAL "x64")
```

File: Core/Node-API/CMakeLists.txt [hunk 4]
```diff
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
...
+                # Enable V8 Pointer Compression
+                # https://v8.dev/blog/pointer-compression
+                # https://stackoverflow.com/q/62921373
+                set(NAPI_DEFINITIONS V8_COMPRESS_POINTERS)
```
Comment (bghgary) [inline Core/Node-API/CMakeLists.txt:139]: Is this private or public?
```suggestion
                set(NAPI_DEFINITIONS PRIVATE V8_COMPRESS_POINTERS)
```

File: Core/Node-API/CMakeLists.txt [hunk 5]
```diff
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
...
+
+            # only 1 imported location per library -> Adding 1 library per file
+            foreach(V8FILE ${V8_DIST} )
+                get_filename_component(V8FILE_NAME "${V8FILE}" NAME_WE)
+                add_library("v8::${V8FILE_NAME}" SHARED IMPORTED)
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8_libbase.dll.lib")
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_LOCATION ${V8FILE})
+            endforeach()
+
+            set(LINK_LIBRARIES v8 v8::icudtl v8::icui18n v8::icuuc v8::v8 v8::v8_libbase v8::v8_libplatform v8::zlib)
```
Comment (bghgary) [inline Core/Node-API/CMakeLists.txt:133]: Should probably follow the pattern of appending to existing and indicate PRIVATE, INTERFACE, PUBLIC.
```suggestion
            set(LINK_LIBRARIES ${LINK_LIBRARIES}
                PRIVATE v8::icudtl
                PRIVATE v8::icui18n
                PRIVATE v8::icuuc
                PRIVATE v8::v8
                PRIVATE v8::v8_libbase
                PRIVATE v8::v8_libplatform
                PRIVATE v8::zlib)
```

File: Tests/UnitTests/CMakeLists.txt [hunk 6]
```diff
@@ -59,7 +59,7 @@ target_link_libraries(UnitTests
 # See https://gitlab.kitware.com/cmake/cmake/-/issues/23543
 # If we can set minimum required to 3.26+, then we can use the `copy -t` syntax instead.
 add_custom_command(TARGET UnitTests POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E $<IF:$<BOOL:$<TARGET_RUNTIME_DLLS:UnitTests>>,copy,true> $<TARGET_RUNTIME_DLLS:UnitTests> $<TARGET_FILE_DIR:UnitTests>)
+    COMMAND ${CMAKE_COMMAND} -E $<IF:$<BOOL:$<TARGET_RUNTIME_DLLS:UnitTests>>,copy,true> $<TARGET_RUNTIME_DLLS:UnitTests> $<TARGET_FILE_DIR:UnitTests> COMMAND_EXPAND_LISTS)
```
Comment (bghgary) [inline Tests/UnitTests/CMakeLists.txt:62]: Why is this needed?

File: Core/Node-API/CMakeLists.txt (hunk 1 above)
Comment (CedricGuillemet) [reply Core/Node-API/CMakeLists.txt:101]: Fixed

File: Core/Node-API/CMakeLists.txt (hunk 2 above)
Comment (CedricGuillemet) [reply Core/Node-API/CMakeLists.txt:126]: Fixed

File: Tests/UnitTests/CMakeLists.txt (hunk 6 above)
Comment (CedricGuillemet) [reply Tests/UnitTests/CMakeLists.txt:62]: list in generator expression is composed of arguments separated by ';' which is not a valid parameter for file copy. That flag replaces the ; by a space making it valid. as the doc says:
```
Lists in COMMAND arguments will be expanded, including those created with [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#manual:cmake-generator-expressions(7)), allowing COMMAND arguments such as ${CC} "-I$<JOIN:$<TARGET_PROPERTY:foo,INCLUDE_DIRECTORIES>,;-I>" foo.cc to be properly expanded.
```

File: Core/Node-API/CMakeLists.txt (hunk 4 above)
Comment (CedricGuillemet) [reply Core/Node-API/CMakeLists.txt]: public!

File: Core/Node-API/CMakeLists.txt (hunk 5 above)
Comment (CedricGuillemet) [reply Core/Node-API/CMakeLists.txt]: Fixed with v8 added as public

File: Core/Node-API/CMakeLists.txt (hunk 3 above)
Comment (CedricGuillemet) [reply Core/Node-API/CMakeLists.txt:142]: Fixed
